cassovary-benchmarks/src/main/scala/com/twitter/cassovary/PerformanceBenchmark.scala

```diff
@@ -93,73 +93,86 @@ object PerformanceBenchmark extends App with GzipGraphDownloader {
-  if (localFileFlag.isDefined) files += ((SMALL_FILES_DIRECTORY, localFileFlag()))
-  if (remoteFileFlag.isDefined) files += cacheRemoteFile(remoteFileFlag())
-  if (files.isEmpty) {
-    println("No files specified on command line. Taking default graph files facebook and wiki-Vote.")
-    files ++= smallFiles
+
+  def cacheRemoteFile(url: String): (String, String) = {
+    printf("Downloading remote file from %s\n", url)
+    new File(CACHE_DIRECTORY).mkdirs()
+    val name = url.split("/").last.split("\\.")(0) + ".txt"
+    val target = CACHE_DIRECTORY + name
+    downloadAndUnpack(url, target)
+    (CACHE_DIRECTORY, name)
   }
-  if (globalPRFlag()) { benchmarks += (g => new PageRankBenchmark(g)) }
-  if (pprFlag()) { benchmarks += (g => new PersonalizedPageRankBenchmark(g)) }
-
-  centFlag() match {
-    case "indegree"  => benchmarks += (g => new InDegreeCentralityBenchmark(g))
-    case "outdegree" => benchmarks += (g => new OutDegreeCentralityBenchmark(g))
-    case "closeness" => benchmarks += (g => new ClosenessCentralityBenchmark(g))
-    case "all"       => benchmarks.append(g => new InDegreeCentralityBenchmark(g),
-      g => new OutDegreeCentralityBenchmark(g), g => new ClosenessCentralityBenchmark(g))
-    case "none" =>
-    case s: String => printf("%s is not a valid centrality option.  Please use indegree, outdegree, closeness or all.\n", s)
+
+  def readGraph(path: String, filename: String, adjacencyList: Boolean): DirectedGraph[Node] = {
+    if (adjacencyList) {
+      AdjacencyListGraphReader.forIntIds(path, filename).toArrayBasedDirectedGraph()
+    } else {
+      val sep = separatorInt().toChar
+      printf("Using Character (%d in Int) as separator\n", sep.toInt)
+      ListOfEdgesGraphReader.forIntIds(path, filename, graphDir = StoredGraphDir.BothInOut,
+        separator = sep).toSharedArrayBasedDirectedGraph(forceSparseRepr = None)
+      //separator = sep).toArrayBasedDirectedGraph(neighborsSortingStrategy = LeaveUnsorted,
+      //      forceSparseRepr = None)
+    }
   }
 
-  if (hitsFlag()) { benchmarks += (g => new HitsBenchmark(g)) }
-  if (getNodeFlag() > 0) { benchmarks += (g => new GetNodeByIdBenchmark(g, getNodeFlag(),
-    GraphDir.OutDir))}
-  if (helpFlag()) {
-    println(flags.usage)
-  } else {
-    def readGraph(path : String, filename : String, adjacencyList: Boolean) : DirectedGraph[Node] = {
-      if (adjacencyList) {
-        AdjacencyListGraphReader.forIntIds(path, filename).toArrayBasedDirectedGraph()
-      } else {
-        val sep = separatorInt().toChar
-        printf("Using Character (%d in Int) as separator\n", sep.toInt)
-        ListOfEdgesGraphReader.forIntIds(path, filename, graphDir = StoredGraphDir.BothInOut,
-          separator = sep).toSharedArrayBasedDirectedGraph(forceSparseRepr = None)
-          //separator = sep).toArrayBasedDirectedGraph(neighborsSortingStrategy = LeaveUnsorted,
-          //      forceSparseRepr = None)
-      }
+  def main(args: Array[String]): Unit = {
```

Comment:
Sbt JMH plugin gets crazy and fails just after start when using App trait, because of using DelayedInit.